apps/admin/core/src/main/java/com/adobe/dx/admin/servlet/EncryptionServlet.java

```diff
@@ -0,0 +1,127 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2020 Adobe
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+
+package com.adobe.dx.admin.servlet;
+
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static org.apache.commons.lang.StringUtils.EMPTY;
+import static org.apache.sling.api.servlets.HttpConstants.METHOD_GET;
+import static org.apache.sling.api.servlets.ServletResolverConstants.SLING_SERVLET_EXTENSIONS;
+import static org.apache.sling.api.servlets.ServletResolverConstants.SLING_SERVLET_METHODS;
+import static org.apache.sling.api.servlets.ServletResolverConstants.SLING_SERVLET_PATHS;
+
+import com.adobe.granite.crypto.CryptoException;
+import com.adobe.granite.crypto.CryptoSupport;
+import com.adobe.xfa.ut.StringUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.servlet.Servlet;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.jetbrains.annotations.NotNull;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(
+    service = Servlet.class,
+    property = {
+        SLING_SERVLET_PATHS + "=/services/private/encryptValues",
+        SLING_SERVLET_EXTENSIONS +  "=json",
+        SLING_SERVLET_METHODS + "=" + METHOD_GET})
+public class EncryptionServlet extends SlingSafeMethodsServlet {
```

Comment:
also, why creating a specific servlet?
granite has that already typically 
``` curl -u admin:admin -F datum="please encrypt me" http://localhost:4502/system/console/crypto.json | jq ```
will get you 

``` 
{
  "protected": "{df4de2f114e4180...7a39753e56b}"
}
```

Reply:
That doesn't work in Skyline because it's behind system/console.

Reply:
i see, couldn't then we proxy the call to that servlet?

Reply:
2 things here, 
1. The OOTB servlet doesn't support multiple values - which this servlet does and we have a use case for, failing which we would have to have multiple calls to the servlet.
2. For a single value, we are just calling CryptoSupport to encrypt - which is again almost the same as the proxy. So don't see a great value add there.